File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.py
class Solution:
    def findMinArrowShots(self, points: List[List[int]]) -> int:
        points.sort()
        answer = 1
        currMax = points[0][1] 
        for i in range(1, len(points)):
            if points[i][0] <= currMax:
                currMax = min(currMax, points[i][1])
            else:
                answer += 1 
                currMax = points[i][1] 
        
        return answer

File: 0452-minimum-number-of-arrows-to-burst-balloons/0452-minimum-number-of-arrows-to-burst-balloons.java
class Solution {
    public int findMinArrowShots(int[][] points) {
        Arrays.sort(points, (a,b) -> Integer.compare(a[0],b[0])); 
        int end = 0, n = points.length; 
        int answer = 0; 
        while(end < n){
            int currEnd = points[end][1]; 
            while(end + 1 < n && points[end+1][0] <= currEnd){
                currEnd = Math.min(currEnd, points[end+1][1]);
                end++; 
            }
            
            answer++; 
            end++; 
        }
        return answer;  
    }
}